kong/plugins/file-log/schema.lua

```diff
@@ -10,8 +28,10 @@ return {
         fields = {
           { path = { type = "string",
                      required = true,
-                     match = [[^[^*&%%\`]+$]],
-                     err = "not a valid filename",
+                     match = path_pattern,
+                     -- to avoid the path traversal attack
+                     not_match = [[%.%.]],
```

Comment:
don't disallow `..` at all, instead, calculate the path based on `..`. For example, if `plugin_file_log_path_prefix` is `/tmp/abc`, then `path` is
`/root/../tmp/abc`, should we allow it?

The other case is for symlinks, usually programs talks to local fs that has path restrictions also have a switch to follow/allow symlink or now. Consider there's a symlink
`/tmp/abc -> /root/abc/`, and `plugin_file_log_path_prefix` is `/tmp`, should you allow plugin write to `/tmp/abc/xxx`?

Reply:
Options:
* Make it always concatenate with the global base. Then canonize (penlight has pl.path.normpath) and make sure that the resulting path is still inside the global base. Drawbacks: custom validations are not great when you are trying to implement a *different CP*.
* Not validate here at all, but make the save operation fail very noisily on the logs.